src/types/mod.rs

```diff
@@ -195,60 +247,54 @@ impl<V: Variable, O: Operator> Context<V, O> {
     //
     // pub fn slice(&self, c: Context<V, O>, t: Term<V, O>) -> Option<Vec<Term<V, O>>>
 }
-impl<V: Variable, O: Operator> From<Term<V, O>> for Context<V, O> {
-    fn from(t: Term<V, O>) -> Self {
+impl From<Term> for Context {
+    fn from(t: Term) -> Context {
         match t {
             Term::Variable(v) => Context::Variable(v),
-            Term::Application { head, args } => {
+            Term::Application { op, args } => {
                 let args = args.into_iter().map(Context::from).collect();
-                Context::Application { head, args }
+                Context::Application { op, args }
             }
         }
     }
 }
```

Comment:
Both here for `Context` and earlier for `Atom`, we'll probably want to add `TryFrom` traits to convert back to `Term`s, `Operator`s, and `Variable`s. That doesn't need to be done here, I'm just marking it as future work.